### ISSUE TITLE
"exec format error"

### ISSUE THREAD

**Issue by viperfx** [dimonomid/geekmarks#7]
Hi, 

I am using OSX and I get the following issue when trying to run the last `make -C server/envs/dev` command

```
make -C ../..
go generate dmitryfrank.com/geekmarks/...
go build --race -i dmitryfrank.com/geekmarks/server/cmd/geekmarks-server
docker build -t docker.io/dimonomid/geekmarks-server .
Sending build context to Docker daemon 16.69 MB
Step 1/3 : FROM golang:1.7
 ---> 7afbc2b03b9e
Step 2/3 : ADD geekmarks-server /
 ---> Using cache
 ---> 1ab74f6e4e90
Step 3/3 : ENTRYPOINT /geekmarks-server
 ---> Using cache
 ---> df07124cca50
Successfully built df07124cca50
rm geekmarks-server
docker-compose  build && docker-compose  up
postgres uses an image, skipping
geekmarks uses an image, skipping
Starting geekmarks-dev-postgres
Starting geekmarks-dev-main
Attaching to geekmarks-dev-postgres, geekmarks-dev-main
geekmarks-dev-main | standard_init_linux.go:178: exec user process caused "exec format error"
geekmarks-dev-main exited with code 1
geekmarks-dev-postgres | LOG:  database system was shut down at 2017-02-21 02:11:08 UTC
geekmarks-dev-postgres | LOG:  MultiXact member wraparound protections are now enabled
geekmarks-dev-postgres | LOG:  database system is ready to accept connections
geekmarks-dev-postgres | LOG:  autovacuum launcher started

```

**Comment by dimonomid** [comment]
Hi,

Unfortunately I don't have any Mac machine to test it on, and from what I know, there are some pecularities in how docker works on Mac. Please see this docker issue: https://github.com/docker/docker/issues/26665 does that help in your case?

You can also consider running geekmarks binary directly, without docker (and you'll also need postgresql daemon running), as @mariusrugan explained in his comment to another (unrelated) issue https://github.com/dimonomid/geekmarks/issues/5#issuecomment-280836910

**Comment by mariusrugan** [comment]
yeah i found the same issue but at that point i was figuring out the correct go/src/ path for the whole thing to actually start building.
i'm looking now but can't find an answer to what is happening right now.

same error
`standard_init_linux.go:178: exec user process caused "exec format error"`
googling for it leads to:
https://github.com/docker-library/mysql/issues/228
- 1. platform incompatibility ? can't be the case 
i'm building with 
`	go generate dmitryfrank.com/geekmarks/...`
`	go build dmitryfrank.com/geekmarks/server/cmd/geekmarks-server`

- 2. issue with Dockerfile entrypoint ?

Cutting now down everything but geekmarks in the make-to-docker up chain; Even with a very flat Dockerfile 

`FROM golang:1.7`
`ADD geekmarks-server /geekmarks-server`

build with `docker build .`
run with `docker run -i -t #IMAGE_HASH# bash`
i get:

`root@fcd6040eeee4:/# ./geekmarks-server
bash: ./geekmarks-server: cannot execute binary file: Exec format error`

maybe @dimonomid you have more experience running it  ?

**Comment by paridhika** [comment]
Hi,
can someone tell me which docker image I need for raspberry pi 3.
linux/arm64, linux/arm/v7, linux/arm/v6
